### PR TITLE
Saddle Finance DEX on Optimism Chain

### PR DESCRIPTION
**Pull request by PatelPrinci**
## Thank you for contributing to Spellbook 🪄
Please open the PR in **draft** and mark as ready when you want to request a review. 

### Description:

[...]

---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: dbt_subprojects/dex/models/trades/optimism/platforms/saddle_finance_optimism_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{{ config(
+    schema = 'saddle_finance_optimism',
+    alias = 'base_trades',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index']
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/optimism/platforms/saddle_finance_optimism_base_trades.sql:8]:
```suggestion
    unique_key = ['tx_hash', 'evt_index']
```

File: dbt_subprojects/dex/models/trades/optimism/_schema.yml [hunk 2]
```diff
@@ -568,3 +568,22 @@ models:
...
+    description: "saddle finance optimism base trades"
+    data_tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - tx_hash
+            - evt_index
+      - check_dex_base_trades_seed:
+          seed_file: ref('saddle_finance_optimism_base_trades_seed') 
+          filter:
+            version: 2
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/optimism/_schema.yml:589]:
```suggestion
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Minor fixes to make sure the seed test working properly.

**Comment by Hosuke** [comment]
Please run `git merge main` to update this PR.
Thank you.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @PatelPrinci

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**